### PR TITLE
colloid: init at 2022-05-15

### PR DESCRIPTION
**Pull request by cirkku**
###### Description of changes

Add Colloid GTK  theme.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Please rebase

File: pkgs/data/themes/colloid/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    HOME=./tmp
+    bash install.sh -d $out/share/themes -t all ${lib.optionalString (tweaks != []) "--tweaks " + builtins.toString tweaks}
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    jdupes -L -r $out/share
```
Comment (Lord-Valen) [inline pkgs/data/themes/colloid/default.nix:50]: Why is this necessary? If their build script is creating duplicates, that should be fixed upstream.

**Review by Lord-Valen — CHANGES_REQUESTED** [review]
This should be updated to the latest version, rebased, and consist of two commits: 
1. `maintainers: add <handle>`
2. `colloid-gtk-theme: init at <version>`

Upstream states that it also depends on https://github.com/vinceliuice/Colloid-icon-theme, so you will probably need a third commit before the gtk-theme commit: `colloid-icon-theme: init at <version>` and to add `colloid-icon-theme` to `buildInputs`.

File: pkgs/data/themes/colloid/default.nix (hunk 1 above)
Comment (cirkku) [reply pkgs/data/themes/colloid/default.nix:50]: Hey! Sorry for keeping this PR open for no reason. I had to deal with real life things and forgot it entirely. There is already an alternative package for this theme in nixpkgs, so there is no need to keep this one around. I believe the reason I added the postInstall is only in the case of installing multiple copies of the theme, in which case some redundant files are removed. I may be wrong, but I believe it should not be applied at the original install script, as it is not opinionated about install locations and other configuration details, whereas with NixOS details about the installation can be reliably made. It should be possible to upstream changes to the install script, but those concerns should be directed to the maintainer of the current nixpkgs package, as it appears to use jdupes as well.
Regarding the depency on colloid icon theme. I believe it is only for the best visual experience and the theme works with alternative icon sets without issues at least in my personal use.
